### PR TITLE
support expression in projection

### PR DESCRIPTION
**Pull request by cheese8**
Refer #14622.

Changes proposed in this pull request:
-support expression in projection

### CODE REVIEW

**Review by strongduanmu — CHANGES_REQUESTED** [review]
@cheese8 Dose this pr support field encrypt and decrypt in concat function?

File: shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/impl/EncryptProjectionTokenGenerator.java [hunk 1]
```diff
@@ -144,7 +192,7 @@ private ColumnProjection buildColumnProjection(final ColumnProjectionSegment seg
     }
     
     private Collection<ColumnProjection> generateProjections(final String tableName, final ColumnProjection column, final SubqueryType subqueryType, final boolean shorthand, 
-                                                             final ShorthandProjectionSegment segment) {
+                                                             final ShorthandProjectionSegment segment, final boolean forceNoneAlias) {
```
Comment (strongduanmu) [inline shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/impl/EncryptProjectionTokenGenerator.java:195]: @cheese8 Why add forceNoneAlias?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Comment by strongduanmu** [comment]
Hi @cheese8, does this pr has any update?

**Comment by cheese8** [comment]
> @cheese8 Dose this pr support field encrypt and decrypt in concat function?

Yes, I'm testing it.

File: shardingsphere-test/shardingsphere-rewrite-test/src/test/resources/scenario/encrypt/case/select_for_query_with_cipher.xml [hunk 2]
```diff
@@ -47,6 +47,11 @@
         <output sql="SELECT account_id, cipher_password AS password, cipher_amount AS a, status AS s FROM t_account WHERE account_id = 1 AND assisted_query_password = 'assisted_query_aaa' AND cipher_amount = 'encrypt_1000' AND status = 'OK'" />
     </rewrite-assertion>
 
+    <rewrite-assertion id="select_for_concat_literals" db-types="MySQL">
+        <input sql="SELECT account_id, concat(account_id, password, amount) as ap, amount AS a, status AS s FROM t_account WHERE account_id = 1 AND password = 'aaa' AND amount = 1000 AND status = 'OK'" />
```
Comment (LeeGuoPing) [inline shardingsphere-test/shardingsphere-rewrite-test/src/test/resources/scenario/encrypt/case/select_for_query_with_cipher.xml:51]: It's better to use uppercase `AS`

**Pull request closed without merging**